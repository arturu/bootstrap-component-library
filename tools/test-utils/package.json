{
  "name": "@openeuropa/bcl-test-utils",
  "version": "0.1.3",
  "description": "A local test helper",
  "main": "index.js",
  "private": true,
  "devDependencies": {
    "deepmerge": "4.2.2"
  }
}
