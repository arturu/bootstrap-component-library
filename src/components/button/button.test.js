import { merge, renderTwigFileAsNode } from "@oe/test-utils";
import demoData from "@oe/data-button/data";

const template = "@oe/button/button.html.twig";
const render = (params) => renderTwigFileAsNode(template, params);
const variants = [
  "primary",
  "secondary",
  "success",
  "danger",
  "warning",
  "info",
  "light",
  "dark",
  "link"
];
const sizes = [
  "lg",
  "md",
  "sm",
];

describe("OE - Button", () => {
  variants.forEach((variant) => {
    test(`${variant} renders correctly`, () => {
      expect.assertions(1);

      return expect(
        render({ ...demoData, variant: variant })
      ).resolves.toMatchSnapshot();
    });
  });

  sizes.forEach((size) => {
    test(`${size} renders correctly`, () => {
      expect.assertions(1);

      return expect(
        render({ ...demoData, size: size })
      ).resolves.toMatchSnapshot();
    });
  });
});

describe("OE - Button Outline", () => {
  variants.forEach((variant) => {
    test(`${variant} renders correctly`, () => {
      expect.assertions(1);

      return expect(
        render({ ...demoData, variant: variant, outline: true })
      ).resolves.toMatchSnapshot();
    });
  });
});
