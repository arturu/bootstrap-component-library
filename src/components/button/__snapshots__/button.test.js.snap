// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`OE - Button Outline danger renders correctly 1`] = `
<button
  class="btn btn-outline-danger btn-md"
  type="button"
>
  Label
</button>
`;

exports[`OE - Button Outline dark renders correctly 1`] = `
<button
  class="btn btn-outline-dark btn-md"
  type="button"
>
  Label
</button>
`;

exports[`OE - Button Outline info renders correctly 1`] = `
<button
  class="btn btn-outline-info btn-md"
  type="button"
>
  Label
</button>
`;

exports[`OE - Button Outline light renders correctly 1`] = `
<button
  class="btn btn-outline-light btn-md"
  type="button"
>
  Label
</button>
`;

exports[`OE - Button Outline link renders correctly 1`] = `
<button
  class="btn btn-outline-link btn-md"
  type="button"
>
  Label
</button>
`;

exports[`OE - Button Outline primary renders correctly 1`] = `
<button
  class="btn btn-outline-primary btn-md"
  type="button"
>
  Label
</button>
`;

exports[`OE - Button Outline secondary renders correctly 1`] = `
<button
  class="btn btn-outline-secondary btn-md"
  type="button"
>
  Label
</button>
`;

exports[`OE - Button Outline success renders correctly 1`] = `
<button
  class="btn btn-outline-success btn-md"
  type="button"
>
  Label
</button>
`;

exports[`OE - Button Outline warning renders correctly 1`] = `
<button
  class="btn btn-outline-warning btn-md"
  type="button"
>
  Label
</button>
`;

exports[`OE - Button danger renders correctly 1`] = `
<button
  class="btn btn-danger btn-md"
  type="button"
>
  Label
</button>
`;

exports[`OE - Button dark renders correctly 1`] = `
<button
  class="btn btn-dark btn-md"
  type="button"
>
  Label
</button>
`;

exports[`OE - Button info renders correctly 1`] = `
<button
  class="btn btn-info btn-md"
  type="button"
>
  Label
</button>
`;

exports[`OE - Button lg renders correctly 1`] = `
<button
  class="btn btn-primary btn-lg"
  type="button"
>
  Label
</button>
`;

exports[`OE - Button light renders correctly 1`] = `
<button
  class="btn btn-light btn-md"
  type="button"
>
  Label
</button>
`;

exports[`OE - Button link renders correctly 1`] = `
<button
  class="btn btn-link btn-md"
  type="button"
>
  Label
</button>
`;

exports[`OE - Button md renders correctly 1`] = `
<button
  class="btn btn-primary btn-md"
  type="button"
>
  Label
</button>
`;

exports[`OE - Button primary renders correctly 1`] = `
<button
  class="btn btn-primary btn-md"
  type="button"
>
  Label
</button>
`;

exports[`OE - Button secondary renders correctly 1`] = `
<button
  class="btn btn-secondary btn-md"
  type="button"
>
  Label
</button>
`;

exports[`OE - Button sm renders correctly 1`] = `
<button
  class="btn btn-primary btn-sm"
  type="button"
>
  Label
</button>
`;

exports[`OE - Button success renders correctly 1`] = `
<button
  class="btn btn-success btn-md"
  type="button"
>
  Label
</button>
`;

exports[`OE - Button warning renders correctly 1`] = `
<button
  class="btn btn-warning btn-md"
  type="button"
>
  Label
</button>
`;

exports[`OE - Button with icon renders correctly 1`] = `
<button
  class="btn btn-primary btn-md"
  type="button"
>
  Label
  <svg
    class="bi icon--s icon--rotate-90"
  >
    <use
      xlink:href="bootstrap-icons.svg#mouse"
    />
  </svg>
</button>
`;

exports[`OE - Button with toggle renders correctly 1`] = `
<button
  aria-controls="collapseExample"
  aria-expanded="false"
  autocomplete="off"
  class="btn btn-primary btn-md"
  data-bs-target="#collapseExample"
  data-bs-toggle="collapse"
  type="button"
>
  I am a link
</button>
`;
